core/templates/services/rte-helper-modal.controller.ts

```diff
@@ -221,7 +232,9 @@ export class RteHelperModalComponent {
     });
 
     this.customizationArgsForm = this.fb.group(formGroupControls);
-
+    if (this.componentId === 'video') {
+      this.currentRteIsVideoEditor = true;
```

Comment:
Instead of keeping these variables, perhaps you could do away with them and just do explicit comparison of componentId?

Reply:
In the places where this.currentRteIsXEditor is used (in this file), do away with that check and use "this.componentId === ..." instead. Drop the superfluous variables.

We are now adding a third component here so we need to find a way to do this more scalably, which is what the discussion up to now has been about. Now that we have a way to know the component ID within this modal controller, let's use it for all the things where there were hacks before, and clean those up. (Often the right time to do this cleanup is either at the outset or when we add the third example of something, since by then it's clearly a repeated pattern.)

Reply:
Ok I think I got this one @seanlip, since we now are aware of componentId we do not need these checks anymore if we use 
```html
  <div class="modal-footer">
    <input type="submit"
           class="btn btn-danger e2e-test-delete-button"
           value="Delete"
           *ngIf="!defaultRTEComponent"
           (click)="delete()">
    <input class="btn btn-secondary e2e-test-cancel-rich-text-editor oppia-cancel-btn"
           value="Cancel"
           (click)="cancel()">
    <input *ngIf="componentId !== 'link' && componentId !== 'math' && componentId !== 'video'"
           class="btn btn-success e2e-test-close-rich-text-component-editor oppia-save-btn"
           value="Done"
           (click)="save()"
           [disabled]="!customizationArgsForm.valid">
    <input *ngIf="componentId === 'link'"
           class="btn btn-success e2e-test-close-rich-text-component-editor oppia-save-btn"
           value="Done"
           (click)="save()"
           [disabled]="!customizationArgsForm.valid || disableSaveButtonForLinkRte()">
    <input *ngIf="componentId === 'math'"
           class="btn btn-success e2e-test-close-rich-text-component-editor oppia-save-btn"
           value="Done"
           (click)="save()"
           [disabled]="!customizationArgsForm.valid || disableSaveButtonForMathRte()">
    <input *ngIf="componentId === 'video'"
           class="btn btn-success e2e-test-close-rich-text-component-editor oppia-save-btn"
           value="Done"
           (click)="save()"
           [disabled]="!customizationArgsForm.valid || disableSaveButtonForVideoRte() || saveButtonIsDisabled">
  </div>
```
in rte-helper-modal.component.html instead, seems reasonable?

Reply:
No, because the general rte-helper-modal is still referring to individual components. You should have a single save button with a [disabled]="!isSaveButtonEnabled()" and the isSaveButtonEnabled() should contain any generic logic needed.

My original comment was relating to the TypeScript, and pointing out that we probably don't need the currentRteIsXEditor variables any more because we can replace them with "componentId == ..." instead.